### PR TITLE
DATAES-422 - Add support for IndicesOptions in search queries.

### PR DESCRIPTION
**Pull request by aturkovic**
See also: [DATAES-191](https://jira.spring.io/browse/DATAES-191)

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by xhaggi** [comment]
@aturkovic could you please rebase your PR against master.

File: src/main/java/org/springframework/data/elasticsearch/core/query/AbstractQuery.java [hunk 1]
```diff
@@ -149,4 +152,12 @@ public void setSearchType(SearchType searchType) {
 	public SearchType getSearchType() {
 		return searchType;
 	}
+
+	public IndicesOptions getIndicesOptions() {
+		return indicesOptions;
+	}
+
+	public void setIndicesOptions(final IndicesOptions indicesOptions) {
```
Comment (xhaggi) [inline src/main/java/org/springframework/data/elasticsearch/core/query/AbstractQuery.java]: none of the parameters are `final` in this class, so please remove it.

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 2]
```diff
@@ -242,6 +244,29 @@ public void shouldReturnPageForGivenSearchQuery() {
 		assertThat(sampleEntities.getTotalElements(), greaterThanOrEqualTo(1L));
 	}
 
+	@Test // DATAES-422 - Add support for IndicesOptions in search queries
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]: please move the comment above `@Test`

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 3]
```diff
@@ -242,6 +244,29 @@ public void shouldReturnPageForGivenSearchQuery() {
...
+		IndexQuery idxQuery = new IndexQueryBuilder().withIndexName(INDEX_1_NAME)
+				.withId(sampleEntity.getId())
+				.withObject(sampleEntity).build();
+
+		elasticsearchTemplate.index(idxQuery);
+		elasticsearchTemplate.refresh(INDEX_1_NAME);
+
+		// when
+		SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(matchAllQuery())
+				.withIndices(INDEX_1_NAME, INDEX_2_NAME).withIndicesOptions(IndicesOptions.lenientExpandOpen()).build();
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]: could you put every `.withXXX` on a new line?

**Comment by aturkovic** [comment]
Fixed as requested, will rebase as soon as I can...

**Comment by aturkovic** [comment]
Sorry, my knowledge of git and doing proper PRs is very limited, appreciate the help...
I'm not sure if I've done it correctly this time?

**Comment by xhaggi** [comment]
@aturkovic no problem, you did it right 👍

**Comment by aturkovic** [comment]
Great, just one note, I had to do `git fetch spring master`, not `git fetch spring/master` in the first step.

**Comment by xhaggi** [comment]
My fault you only need to specify the remote `git fetch spring`

**Pull request merged**